api/src/main/java/io/grpc/GlobalInterceptors.java

```diff
@@ -0,0 +1,144 @@
+   *     Interceptors.
+   * @param serverInterceptorList list of {@link ServerInterceptor} that make up global Server
+   *     Interceptors.
+   * @param serverStreamTracerFactoryList list of {@link ServerStreamTracer.Factory} that make up
+   *     global ServerStreamTracer factories.
+   */
+  public static void setInterceptorsTracers(
+      List<ClientInterceptor> clientInterceptorList,
+      List<ServerInterceptor> serverInterceptorList,
+      List<ServerStreamTracer.Factory> serverStreamTracerFactoryList) {
+    try {
+      GlobalInterceptors.getInstance()
+          .setGlobalInterceptorsTracers(
+              clientInterceptorList, serverInterceptorList, serverStreamTracerFactoryList);
+    } catch (Exception exception) {
+      throw exception;
+    }
+  }
+
+  /** Returns the list of global {@link ClientInterceptor}. If not set, this returns null. */
+  public static List<ClientInterceptor> getClientInterceptors() {
+    return GlobalInterceptors.getInstance().getGlobalClientInterceptors();
+  }
+
+  /** Returns list of global {@link ServerInterceptor}. If not set, this returns null. */
+  public static List<ServerInterceptor> getServerInterceptors() {
+    return GlobalInterceptors.getInstance().getGlobalServerInterceptors();
+  }
+
+  /** Returns list of global {@link ServerStreamTracer.Factory}. If not set, this returns null. */
+  public static List<ServerStreamTracer.Factory> getServerStreamTracerFactories() {
+    return GlobalInterceptors.getInstance().getGlobalServerStreamTracerFactories();
+  }
+
+  @VisibleForTesting
+  synchronized void setGlobalInterceptorsTracers(
+      List<ClientInterceptor> clientInterceptorList,
+      List<ServerInterceptor> serverInterceptorList,
+      List<ServerStreamTracer.Factory> serverStreamTracerFactoryList) {
+    if (isGlobalInterceptorsTracersGet) {
+      throw new IllegalStateException("Set cannot be called after any corresponding get call");
+    }
+    if (isGlobalInterceptorsTracersSet) {
+      throw new IllegalStateException("Global interceptors and tracers are already set");
+    }
+
+    if (clientInterceptorList != null) {
+      clientInterceptors =
+          ImmutableList.<ClientInterceptor>builder().addAll(clientInterceptorList).build();
+    }
+
+    if (serverInterceptorList != null) {
+      serverInterceptors =
+          ImmutableList.<ServerInterceptor>builder().addAll(serverInterceptorList).build();
+    }
+
+    if (serverStreamTracerFactoryList != null) {
+      serverStreamTracerFactories =
+          ImmutableList.<ServerStreamTracer.Factory>builder()
+              .addAll(serverStreamTracerFactoryList)
+              .build();
+    }
+    isGlobalInterceptorsTracersSet = true;
+  }
+
+  @VisibleForTesting
+  synchronized List<ClientInterceptor> getGlobalClientInterceptors() {
+    if (!isGlobalInterceptorsTracersGet) {
+      isGlobalInterceptorsTracersGet = true;
```

Comment:
Any harm in setting this unconditionally? You will be replacing an unconditional read  with an unconditional write of the same variable - much simpler and not any more inefficient.